flink-core/src/main/java/org/apache/flink/api/connector/sink/SinkWriter.java

```diff
@@ -46,6 +47,14 @@
      */
     void write(InputT element, Context context) throws IOException;
 
+    /**
+     * Add a watermark to the writer.
```

Comment:
Maybe we should explicitly add that this operation is purely optional (yes it can be inferred through `default`). I'm assuming that most sinks will not implement this function. Same for `SinkFunction`.